roles/roles_common.js

```diff
@@ -1231,6 +1231,52 @@ _.extend(Roles, {
   },
 
   /**
+   * By this method you can find out if a role is parent of another role.
+   *
+   * WARNING: If you check this on the client, please make sure all roles are published.
+   *
+   * @method isParentOf
+   * @param {String} roleName The role you have
+   * @param {String} demandedRoleName The role you want to find within roleName
+   * @private
+   * @static
+   */
+  isParentOf: function (roleName, demandedRoleName) {
+    Roles._checkRoleName(roleName);
+    Roles._checkRoleName(demandedRoleName);
+
+    if (roleName === null || demandedRoleName === null) {
+      return false;
+    }
+
+    if (roleName === demandedRoleName) {
+      return true;
+    }
+
+    var alreadyCheckedRoles = [];
+    var rolesToCheck = [roleName];
+    while (rolesToCheck.length !== 0) {
+      var checkRoleName = rolesToCheck.pop();
+      alreadyCheckedRoles.push(checkRoleName);
+
+      var checkRole = Meteor.roles.findOne({ _id: checkRoleName });
+
+      // This should not happen, but this is a problem to address at some other time.
+      if (!checkRole) continue;
+
+      var childRoleIds = _.pluck(checkRole.children, '_id');
+
+      if (_.contains(childRoleIds, demandedRoleName)) {
+        return true;
+      }
+
+      rolesToCheck = _.union(rolesToCheck, _.difference(childRoleIds, alreadyCheckedRoles));
```

Comment:
I think you can just push all children names to `rolesToCheck`, and then at the beginning of the loop you just do `if (checkRoleName === demandedRoleName) return true`. You can then even remove `if (roleName === demandedRoleName) {` outside of the loop.